app/controllers/examples_controller.rb

```diff
@@ -3,13 +3,18 @@ class ExamplesController < ApplicationController
 
   def index
     @examples = Example.includes(:grammar)
+    if params[:grammar_id]
+      @grammar = Grammar.find(params[:grammar_id])
+      @examples = @grammars.where(grammar: @grammar) 
+    end
   end
 
   def show
   end
 
   def new
     @example = Example.new(grammar_id: params[:grammar_id])
+    @grammar = Grammar.find(params[:grammar_id])
```

Comment:
Instead of adding this here you can just do `@example.grammar` in the `show.html.erb`